### DISCUSSION TITLE
uInverter Phase A and C swapped?

### DISCUSSION THREAD

**Discussion by elsevers** [Severson-Group/AMDC-Hardware#222] [Q&A] [answered]
# Summary of Issue and Request

I am experimenting with a `rev A` uInverter board and have discovered that the phase A and phase C silkscreen labels are inconsistent (sometimes swapped) on the PCB. 

@purohitparth25, was this issue corrected for `rev B`? I do not see it referenced in #208. 

One of the students using the `rev B` board for current regulation (@anirudhupadhyaya, @mimayberry, @shamagvasisht): could one of you check to see if this is also an issue on the `rev B` uInverter board?

## Details
When I output a PWM signal on channel 0, 
``` C
pwm_set_duty(0, 0.5);
```
I can use an oscilloscope to confirm that the uInverter testpoint `PWM-A` sees the correct waveform. However, `TP2-Va` does not see the waveform, nor does the PWM pin on the gate driver on the section of the board labeled `Phase A`. Instead, the section labeled `Phase C` sees the waveform.

**Comment by anirudhupadhyaya** [comment]
@elsevers I can confirm that phase A and phase C are swapped in my rev B uInverter. Somehow I missed reporting this. I observed this when I first tried current regulation. My currents would saturate because of the incorrect phase mapping between my control algorithm and  physical hardware.

**Comment by purohitparth25** [comment, marked as answer]
@elsevers @anirudhupadhyaya 
That was related to the identical schematic sheet names which resulted into random mapping between the PWM inputs and the phases (Gate driver IC control signal; for e.g. `PWM A` got connected to Gate driver of `phase C`). See change no. 3 -  `3. Fix sheet designator names in the main schematic page` in this [rev C PR](https://github.com/Severson-Group/AMDC-Hardware/pull/220) for changes made to fix this. 

Now the probed signals at the `PWMA` testpoint should match with the signal at the output of phase A,  `TP1A`.Screenshots from `rev C` PCB document (see the highlighted `net`):
Phase A:
![image (3)](https://user-images.githubusercontent.com/90465550/205819749-8799b747-21c7-49c0-8fde-9d1b731283ca.png)
Phase C:
![image (2)](https://user-images.githubusercontent.com/90465550/205819805-ccf84eac-5340-4b02-983d-c785ed265d58.png)
